mirrord-layer/src/file/ops.rs

```diff
@@ -0,0 +1,255 @@
+use std::{ffi::CString, io::SeekFrom, os::unix::io::RawFd, path::PathBuf};
+
+use libc::{c_int, c_uint, FILE, O_CREAT, O_RDONLY, S_IRUSR, S_IWUSR, S_IXUSR};
+use mirrord_protocol::{
+    CloseFileResponse, OpenFileResponse, OpenOptionsInternal, ReadFileResponse, SeekFileResponse,
+    WriteFileResponse,
+};
+use tokio::sync::oneshot;
+use tracing::{debug, error};
+
+use crate::{
+    common::{
+        CloseFileHook, HookMessage, OpenFileHook, OpenRelativeFileHook, ReadFileHook, SeekFileHook,
+        WriteFileHook,
+    },
+    error::LayerError,
+    file::{OpenOptionsInternalExt, OPEN_FILES},
+    HOOK_SENDER,
+};
+
+#[inline]
```

Comment:
Why did you inline it?

Reply:
Good catch, saw it as a small function, so I just automatically went to `inline`, but it doesn't really make sense there.